crates/ide_completion/src/render/function.rs

```diff
@@ -135,7 +135,15 @@ pub(super) fn add_call_parens<'b>(
                             let ref_ = ref_of_param(ctx, text, param.ty());
                             f(&format_args!("${{{}:{}{}}}", index + offset, ref_, text))
                         }
-                        None => f(&format_args!("${{{}:_}}", index + offset,)),
+                        None => {
+                            let name = match param.ty().as_adt() {
+                                Some(adt) => {
+                                    to_lower_snake_case(&adt.name(ctx.db).as_text().unwrap())
+                                }
+                                None => "_".to_string(),
+                            };
+                            f(&format_args!("${{{}:{}}}", index + offset, name))
+                        }
```

Comment:
instead of `unwrapping` can we do something like `adt.name(ctx.db).as_smol_str().map(|it| to_lower_snake_case(&it))` and if that is none also fallback to the `"_"`?
We should also replace the `"_".to_string()` with `SmolStr::new_inline("_")` (which works with the `as_smol_str` used in my snippet above), that allows us to get rid of some allocations.

Reply:
ah wait forget what I said, I forgot about `to_lower_snake_case` allocating a `String`